service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/OrderRemovalActions.java

```diff
@@ -6,27 +6,28 @@
 import uk.gov.hmcts.reform.fpl.exceptions.removeorder.RemovableOrderActionNotFoundException;
 import uk.gov.hmcts.reform.fpl.model.interfaces.RemovableOrder;
 
-import java.util.UUID;
 import java.util.stream.Stream;
 
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class OrderRemovalActions {
-    private final CMOOrderRemovalAction cmoOrderRemovalAction;
+    private final CMORemovalAction cmoOrderRemovalAction;
     private final GeneratedOrderRemovalAction generatedOrderRemovalAction;
+    private final SDORemovalAction sdoOrderRemovalAction;
 
-    public OrderRemovalAction getAction(UUID removedOrderId,
-                                        RemovableOrder removableOrder) {
+    public OrderRemovalAction getAction(RemovableOrder removableOrder) {
         return getActions()
             .filter(orderRemovalAction -> orderRemovalAction.isAccepted(removableOrder))
             .findFirst()
-            .orElseThrow(() -> new RemovableOrderActionNotFoundException(removedOrderId));
+            .orElseThrow(() -> new RemovableOrderActionNotFoundException(cmoOrderRemovalAction.getClass()
```

Comment:
that desn't look right. It seems to me that was correct in the previous version

Reply:
Looks like it should be the order rather than the action, maybe it would be better to send the class rather than the string representation of it. The exception can then determine what it wants to do with the class when generating the message.
i.e.
```java
            .orElseThrow(() -> new RemovableOrderActionNotFoundException(removableOrder.getClass())
```